wallet/thread/helpers.go

```diff
@@ -0,0 +1,64 @@
+package thread
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/textileio/textile-go/crypto"
+	"github.com/textileio/textile-go/repo"
+	"github.com/textileio/textile-go/wallet/model"
+	"github.com/textileio/textile-go/wallet/util"
+	libp2pc "gx/ipfs/QmaPbCnUMBohSGo3KnxEa2bHqyJVVeEEcwtqJAYxerieBo/go-libp2p-crypto"
+)
+
+// GetBlockDataKey returns the decrypted AES key for a block
+func (t *Thread) GetBlockDataKey(block *repo.Block) ([]byte, error) {
+	if block.Type != repo.PhotoBlock {
+		return nil, errors.New("incorrect block type")
+	}
+	key, err := t.Decrypt(block.DataKeyCipher)
+	if err != nil {
+		log.Errorf("error decrypting key cipher: %s", err)
+		return nil, err
+	}
+	return key, nil
+}
+
+// GetBlockData cats file data from ipfs and tries to decrypt it with the provided block
+func (t *Thread) GetBlockData(path string, block *repo.Block) ([]byte, error) {
+	key, err := t.GetBlockDataKey(block)
+	if err != nil {
+		return nil, err
+	}
+	cipher, err := util.GetDataAtPath(t.ipfs(), path)
+	if err != nil {
+		log.Errorf("error getting file data: %s", err)
+		return nil, err
+	}
+	return crypto.DecryptAES(cipher, key)
+}
+
+// GetFileDataBase64 returns file data encoded as base64 under an ipfs path
+func (t *Thread) GetBlockDataBase64(path string, block *repo.Block) (string, error) {
+	data, err := t.GetBlockData(path, block)
+	if err != nil {
+		return "error", err
+	}
+	return libp2pc.ConfigEncodeKey(data), nil
+}
+
+// GetPhotoMetaData returns photo metadata under an id
+func (t *Thread) GetPhotoMetaData(id string, block *repo.Block) (*model.PhotoMetadata, error) {
```

Comment:
Should this one be exposed to Mobile or is there a diff method to grab the meta file I should use?

Reply:
you can use the generic `GetPhotoData` method... but perhaps we want to make it a bit more explicit for all things in mobile since from a client perspective it has no idea how the data is structures, so req'ing thing like `/photo` are a bit weird.

Reply:
We could do:
`GetPhotoData`
`GetThumbData`
`GetMetaData`

Reply:
That would actually help with the gif thing too.

Reply:
relates to this, https://github.com/textileio/textile-go/issues/170

Reply:
ok, done. image methods will now return the correct data url prefix as well. `TextileImage` will have to be tweaked (delete the prefix: https://github.com/textileio/textile-mobile/blob/ea6addf98ef3bfa3daeeb135d1bf34626f5d5218/ios/NativeViews/TextileImageView.m#L65)